### ISSUE TITLE
[FEATURE REQUEST] Remove/exclude unwanted spawnpoints from db/scan

### ISSUE THREAD

**Issue by PhilKloppers** [RocketMap/RocketMap#1306]
It would be useful to be able to remove (or exclude) unwanted spawnpoints from the db. This could be implemented in the spawnpoint popup similar to the mon popup.

This would reduce the number of scans needed by getting rid of points that aren't travelled to/can't be reached/whatever reason.

**Comment by PhDeez** [comment]
I think this would be harder to implement than you're expecting.  Just because you "exclude" monsters from the map they are still written to the database, just not displaying on your map.  IMO, if you have spawnpoints that are not reachable, why not remove them from the .json file?

**Comment by averylarry** [comment]
Duplicate?  of #1178.

**Comment by PhilKloppers** [comment]
It is possibly related to #1178, but the data source is not an external file. This could be used to solve that problem by doing manual scans with a small step size along a route.

My use case for this request is for spawnpoint scanning using the db, not an external file. Running 1 searchable hex grid search process with a step size of 2 or 3, and a second spawnpoint scan process with -ns and a very large step size (100+) results in a search pattern where spawnpoints discovered by the hex search are dynamically added to the spawnpoint scan. This scan method is not possible using an external spawnpoint file. It does, however, also include spawnpoints which may not be necessary to scan , so a method to remove those points from the db would be useful in decreasing the search queue. Once the required points are in the db, it could then be exported to a file if that's what someone wants.

To be clear, I'm not proposing hiding scanned mons from specific spawnpoints, but excluding the spawnpoints from being scanned entirely.

I'm assuming db spawnpoints are derived from the pokemon table? My proposed implementation would be a 'Remove' link in the spawnpoint popup on the map that would execute a SQL query to remove the records associated with that point - something like a `DELETE * FROM pokemon WHERE spawnpoint_id = "xxxx" LIMIT 1` if spawnpoint_id is unique to the point, or `DELETE * FROM pokemon WHERE latitude = "xxxx" AND longitude = "xxxx" LIMIT 1` if that is not the case.

**Comment by jchristi** [comment]
Don't let the feature request dictate the implementation. If you want spawn points in the db excluded, the solution is simple: add a column to the spawnpoint table called `enabled` and set it to `0` or `1` and add `enabled=1` to (almost) all select queries. This way, the spawnpoints won't constantly be re-added to the database when they are "re-discovered" while scanning.

**Comment by deirdresm** [comment]
"add a column to the spawnpoint table"

Unless I'm missing something, there is no spawnpoint table.

I've actually considered doing a spawnpoint table for a different reason as that could make it easier to segment workers for maps, so: spawnpoint_id, group_id (for grouping spawnpoints), and a boolean disable_spawnpoint (null or false if enabled, true if disabled).

Then when you --dumpspawnpoints, you could limit it to a group_id and enabled.

Use case: my city is shaped so that four large hexes can incorporate it, but I'd like to exclude spawnpoints that are inaccessible or outside the bounds of the city. Plus, assigning spawnpoints to groups means that you don't duplicate efforts for spawnpoints on boundaries between hexes.

**Comment by PhDeez** [comment]
@deirdresm
You can do this now by launching multiple instances of the server with seperate spawn point files and account files.  When you dump your spawn points just specify 1/4 of the entire area.  Break your accounts file in to 4 equal groups and viola... exactly what you're asking for (seperate workers, separate scanning areas and separate workers)

**Comment by jchristi** [comment]
@deirdresm I assumed there was a spawnpoints table since you can view spawnpoints but it must be pulling the data from the pokemon table which is the spawn encounters table that includes spawnpoint id. So a new would be required which has spawnpoint id and enabled flag. PhDeez suggestion will work, but it is a bit clunky.

**Comment by PhDeez** [comment]
@jchristi Well, What I did was run the Tool "Hex-Beehive-Generator" to create my command line entries and lat/lon coordinates, then took those coordinates and scanned each area in "normal" scanning mode for 2 hrs each to make sure  I got all the spawn points captured.  Then, used the --dump-spawnpoints command to create my json files, and THEN used the spawnpoint-clustering tool to reduce worker thread usage.  By doing all this, I can drive to about anywhere within 15m with 7 12 step hexes and only 14 accounts.

IMO, the most work is making sure you capture all of your spawnpoints, and really that's just using your server in normal walk mode.

The only drawback in my opinion is not being able to turn the server off from the webpage, I have to login to the server and kill the python threads.

my shell file looks like this:

nohup python runserver.py -os -l 'lat,lon' &
sleep 5; nohup python runserver.py -ns -enc -mf 5 -sn hex1 -ss hex1.json -l 'lat,lon' -ac account1.csv &
sleep 5; nohup python runserver.py -ns -enc -mf 5 -sn hex2 -ss hex2.json -l 'lat,lon' -ac account2.csv &
sleep 5; nohup python runserver.py -ns -enc -mf 5 -sn hex3 -ss hex3.json -l 'lat,lon' -ac account3.csv &
sleep 5; nohup python runserver.py -ns -enc -mf 5 -sn hex4 -ss hex4.json -l 'lat,lon' -ac account4.csv &
sleep 5; nohup python runserver.py -ns -enc -mf 5 -sn hex5 -ss hex5.json -l 'lat,lon' -ac account5.csv &
sleep 5; nohup python runserver.py -ns -enc -mf 5 -sn hex6 -ss hex6.json -l 'lat,lon' -ac account6.csv &
sleep 5; nohup python runserver.py -ns -enc -mf 5 -sn hex7 -ss hex7.json -l 'lat,lon' -ac account7.csv &
